src/languages/lib/css-shared.js

```diff
@@ -17,7 +17,7 @@ export const MODES = (hljs) => {
       scope: 'selector-attr',
       begin: /\[/,
       end: /\]/,
-      illegal: '$',
+      illegal: /[$]/,
```

Comment:
This is not equivalent and therefore not a _correct_ fix.  One matches end of line (which is what we want)... your new version matches the literal `$` character...

This is actually a bug that must be fixed in the parser itself, not any particular grammar.